osu.Game/Tests/Visual/OsuTestScene.cs

```diff
@@ -70,7 +70,22 @@ protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnl
             Beatmap.SetDefault();
 
             Ruleset = Dependencies.Ruleset;
-            Ruleset.SetDefault();
+
+            var definedRuleset = CreateRuleset()?.RulesetInfo;
+
+            if (definedRuleset != null)
+            {
+                // re-enable the bindable in case it was disabled.
+                // happens when restarting current test scene.
+                Ruleset.Disabled = false;
+
+                // Set global ruleset bindable to the ruleset defined
+                // for this test scene and disallow changing it.
+                Ruleset.Value = definedRuleset;
+                Ruleset.Disabled = true;
+            }
```

Comment:
What's this logic for? Is there a particular scenario that needs this?

If we do decide to have something like this then maybe `OsuTestScene` should take a lease of this?

Reply:
Why was this resolved without comment? I'd also like to know the reason for this.

Reply:
I've let the treatment of this bindable be the same as `Beatmap`'s, and since it's only used in `PlayerTestScene` due to that, I don't think I really need to disable it.

Reply:
If nothing relies on this logic and it was added "just because" then I'd be in favour of removing it.

Reply:
This logic was actually supposed to be removed at the time of my above comment, now I made sure I actually removed it, resolving.